dotnet/src/Connectors/Connectors.OpenAI/Services/OpenAIChatCompletionService.cs

```diff
@@ -71,17 +71,25 @@ public OpenAIChatCompletionService(
         var providedEndpoint = endpoint ?? httpClient?.BaseAddress;
         if (providedEndpoint is not null)
         {
-            // As OpenAI Client automatically adds the chat completions endpoint, we remove it to avoid duplication.
-            const string PathAndQueryPattern = "v1/chat/completions";
-            var providedEndpointText = providedEndpoint.ToString();
-            int index = providedEndpointText.IndexOf(PathAndQueryPattern, StringComparison.OrdinalIgnoreCase);
-            if (index >= 0)
+            // If the provided endpoint does not provide a path, we add a version to the base path for compatibility
+            if (providedEndpoint.PathAndQuery.Length == 0 || providedEndpoint.PathAndQuery == "/")
             {
-                internalClientEndpoint = new Uri($"{providedEndpointText.Substring(0, index)}{providedEndpointText.Substring(index + PathAndQueryPattern.Length)}");
+                internalClientEndpoint = new Uri(providedEndpoint, "/v1/");
             }
             else
             {
-                internalClientEndpoint = providedEndpoint;
+                // As OpenAI Client automatically adds the chatcompletions endpoint, we remove it to avoid duplication.
```

Comment:
Have we always done this?

Correctly a bad endpoint leaves us open to having to continue to do this as things change and also do this consistently across all of our connectors.

Consider using the provided endpoint and expecting developers to get this correct.